### PR TITLE
BREAKING CHANGE: split out high and low res images

### LINKED ISSUE #1173: Card should not be involved with image url manipulation
**Do you want to request a _feature_ or report a _bug_?**

Refactor card so that the consumer of any images passed into card manage the image url, and card does not.

### PR DESCRIPTION
**Pull request by craigbilner**
While relatively large, essentially this PR changes our component APIs to allow more fine grained control over how we handle images and push the image optimisations into the image components. The consumers of the `Image` component still need to decide what `high resolution` and `low resolution` images mean to them.

In our current working example, `ArticleList` decides to load every image with a low resolution size of `100` and only the width of the page for list items that are in the viewport if the browser supports the `IntersectionObserver`. This allows those browsers to provide a faster experience on poor networks/devices. For browsers without an `IntersectionObserver`, the low/high resolution images are still rendered, but for the whole page rather than what is in the viewport.

A future enhancement is also needed to make the `highRes` decision above even more intelligent rather than just asking for a massive image when in reality the list has a small image left-aligned.

This PR highlights areas which are currently not providing an optimal image experience such as `Article`, `Related Articles` and `Native`. Future PRs should address these points.

In addition some UX improvements have been made to make the list pages "less flashy".

## tl;dr

* make high and low res images first class (make it obvious article/related articles/native don't respect it etc.) (edited)
* allow web images to use desired size to provide retina sizes etc in the future
* smooth transition of text and images with no "flashes"
* remove url logic from Card
* standardardise uri (edited)
* fix browsers with no IntersectionObserver

## changes

BREAKING CHANGE: split out high and low res images
refactor: split out fade-in effect
chore: fix meta
fix: no observer SSR

close #1173

### CODE REVIEW

File: packages/card/__tests__/android/__snapshots__/card-with-style.android.test.js.snap [hunk 1]
```diff
@@ -23,7 +23,10 @@ exports[`1. card default state 1`] = `
   >
     <Image
       aspectRatio={0.6666666666666666}
-      uri="https://img.io/img&resize=360"
+      fadeImageIn={false}
```
Comment (colincclark) [inline packages/card/__tests__/android/__snapshots__/card-with-style.android.test.js.snap]: strangely, card is not showing coverage of tests in the CLI so cannot see what it is on this branch

File: packages/image/__tests__/web/__snapshots__/image.web.test.js.snap [hunk 2]
```diff
@@ -1,69 +1,299 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`1. default layout 1`] = `
+.c0 {
```
Comment (colincclark) [inline packages/image/__tests__/web/__snapshots__/image.web.test.js.snap]: odd, cannot see image coverage either

File: packages/article-list/src/article-list.web.js [hunk 3]
```diff
@@ -43,6 +44,19 @@ class ArticleList extends Component {
     );
   }
 
+  componentDidMount() {
+    const newState = {
+      fadeImageIn: true
+    };
+
+    if (typeof window !== "undefined" && !this.observer) {
+      newState.fallback = normaliseWidth(window.clientWidth);
```
Comment (colincclark) [inline packages/article-list/src/article-list.web.js]: presumably, the idea is to split out lazy loading to another package at a later date still?

File: packages/related-articles/src/related-article-item.js [hunk 4]
```diff
@@ -59,11 +59,12 @@ const RelatedArticleItem = ({
     <Link onPress={e => onPress(e, { url: article.url })} url={url}>
       <Card
         contentContainerClass={contentContainerClass}
-        image={{ uri: imageUri }}
         imageContainerClass={imageContainerClass}
         imageRatio={imageRatio}
         imageStyle={imageStyle}
+        imageUri={imageUri}
         isReversed={isReversed}
+        lowResSize={100}
```
Comment (colincclark) [inline packages/related-articles/src/related-article-item.js]: because related articles does not lazy load, the images are low res and blurry on the article page :(

**Comment by colincclark** [comment]
Native seems OK, but I am not sure what a related article image would look like at the bottom of an article page.

File: packages/article-list/src/article-list.web.js (hunk 3 above)
Comment (craigbilner) [reply packages/article-list/src/article-list.web.js]: yep, will pick this up after mocking :+1:

File: packages/card/__tests__/android/__snapshots__/card-with-style.android.test.js.snap (hunk 1 above)
Comment (craigbilner) [reply packages/card/__tests__/android/__snapshots__/card-with-style.android.test.js.snap]: I run `yarn test:android --coverage`? You have to use the `--no-cache` flag if you have previous jest caches for that package

File: packages/related-articles/src/related-article-item.js (hunk 4 above)
Comment (craigbilner) [reply packages/related-articles/src/related-article-item.js]: yep, was done on purpose to make sure this gets picked up with the article work. We should be fetching high res images on mobweb as they'll just be the placeholders otherwise and/or waste bandwidth if never look at
